pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -483,6 +485,9 @@ private void processStreamEvents(MessageBatch messagesAndOffsets, long idlePipeS
               messagesAndOffsets.getMessageLengthAtIndex(index), reuse);
       if (decodedRow != null) {
         try {
+          if (_complexTypeTransformer != null) {
+            decodedRow = _complexTypeTransformer.transform(decodedRow);
```

Comment:
this should just be part of the RecordTransformer returned by the CompositeTransformer#getDefaultTransformer rt? Why create a separate one? If it's separate, future developers always need to take the call on where to include this in the sequence. 
Also, we might introduce pluggable RecordTransformers. If this one is kept separate, it might cause confusion.

Reply:
Because this transformer can add the column of `MULTIPLE_RECORDS_KEY `, which is used in the condition in the next line. 

I feel this complex-type transformer is special (at least in current implementation) due to this, and all other transformers output 1 row only.